source/features/conversation-navigation.tsx

```diff
@@ -0,0 +1,258 @@
+
+	// Coming from another issue with query in URL
+	if (!pageDetect.isConversationList(referrerUrl)) {
+		return parseConversationListURL(new URL(location.href));
+	}
+
+	// Default conversation lists, even global, don't have query in URL
+	// But it seems there is some refined-github feature that adds a default query in URL based on query input
+	// Make sure we have some query as a fallback based on current GitHub defaults
+	const query = referrerUrl.searchParams.get('q');
+
+	if (pageDetect.isGlobalConversationList(referrerUrl)) {
+		// Is global pull requests list
+		if (referrerUrl.pathname.split('/', 2)[1] === 'pulls') {
+			referrerUrl.searchParams.set('q', query ?? `is:open is:pr author:${getUsername()} archived:false`);
+		}
+
+		// Is global issues list
+		if (referrerUrl.pathname.split('/', 2)[1] === 'issues') {
+			referrerUrl.searchParams.set('q', query ?? `is:open is:issue author:${getUsername()} archived:false`);
+		}
+	}
+
+	if (pageDetect.isRepoConversationList(referrerUrl)) {
+		if (pageDetect.isRepoIssueList(referrerUrl)) {
+			referrerUrl.searchParams.set('q', `${query ?? 'is:issue is:open'} repo:${getRepoURL()}`);
+		}
+
+		if (pageDetect.isRepoPRList(referrerUrl)) {
+			referrerUrl.searchParams.set('q', `${query ?? 'is:pr is:open'} repo:${getRepoURL()}`);
+		}
+	}
+
+	return parseConversationListURL(referrerUrl);
+}
+
+function getConversationSearchQuery(first: number, after: string, query: string): string {
+	return `
+search(
+	type: ISSUE,
+	first: ${first},
+	after: "${after}",
+	query: "${query}"
+) {
+	edges {
+		cursor
+		node {
+			... on Issue { number, url }
+			... on PullRequest { number, url }
+		}
+	}
+}
+	`;
+}
+
+function getConversationNumber(): number {
+	const conversationHashtag = select('.gh-header-number')?.textContent ?? '#';
+	return Number.parseInt(conversationHashtag.replace('#', ''), 10);
+}
+
+function parseConversationListURL(url: URL) {
+	return {
+		query: url.searchParams.get('q') ?? '',
+		page: Number.parseInt(url.searchParams.get('page') ?? '1', 10)
+	};
+}
+
+/**
+ * Get a GitHub GraphQL cursor based on offset number
+ * GitHub items cursors have this format e.g. `cursor:1` and are in base64 format
+ * First cursor has offset 1, if offset bellow 1 will be used, still correct first items will be in response but with cursor based on given negative offset
+ *
+ * @param {number} offset
+ * @returns {string}
+ */
+function getCursor(offset: number): string {
+	return btoa(`cursor:${offset}`);
+}
+
+function getOffset(cursor: string): number {
+	return Number.parseInt(atob(cursor).replace('cursor:', ''), 10);
+}
+
+function getPreviousCursor(cursor: string) {
+	return getCursor(getOffset(cursor) - 1);
+}
+
+function getNextCursor(cursor: string) {
+	return getCursor(getOffset(cursor) + 1);
+}
+
+function getPageCursors(page: number) {
+	return {
+		after: getCursor((page - 1) * ITEMS_PER_PAGE),
+		before: getCursor((page * ITEMS_PER_PAGE) + 1)
+	};
+}
```

Comment:
All of this for example should be at most:

```js
function getCursor(offset: number): string {
  	return btoa(`cursor:${offset}`);
  }
 
  function alterCursor(cursor: string, diff: number): string {
  	const index = Number.parseInt(atob(cursor).replace('cursor:', ''), 10);
  	return getCursor(index + diff);
  }
```

Then just call `alterCursor(cursor, diff)`, where diff is -1 or +1

Also I think almost everywhere you return an object, it's making things unnecessarily lengthy